api/services/ProductService.ts

```diff
@@ -0,0 +1,151 @@
+import { ServiceOptions, Service, BaseError } from 'ts-framework-common';
+import Product from '../models/Product';
+import User from '../models/User';
+import ProductStorage from '../models/ProductStorage';
+import AssetService from './AssetService';
+import { Transaction } from 'typeorm';
+import { Extract } from '../models';
+
+export interface ProductServiceOptions extends ServiceOptions{
+
+}
+
+export default class ProductService extends Service {
+    protected static instance: ProductService;
+
+    @Transaction()
+    public async buyProduct(buyerId:string, sellerId:string, productName:string, quantity:number) {
+        const buyer: User =  await User.findById(buyerId);
+        const seller: User =  await User.findById(sellerId);
+        const product: Product = await Product.findByName(productName);
+        const productStorage: ProductStorage = await ProductStorage.findProductStorageForUserAndProduct(seller, product);
+        const assetService: AssetService = AssetService.getInstance({});
+
+        if(quantity <= 0){
+            throw new BaseError('Product quantity is invalid.', {quantity: quantity});
```

Comment:
```suggestion
            throw new BaseError('Product quantity is invalid.', { quantity });
```
Você pode usar shorthand quando o nome da propriedade não muda